From 66336833dbcff38eb7c26595ccc15f7260450cba Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B6rn=20Kottmann?= <joern@apache.org>
Date: Sat, 31 Dec 2016 19:32:48 +0100
Subject: [PATCH] Correctly check for model compatibility

The major versions must match
The last two minor versions should be supported

See issue OPENNLP-906
---
 .../src/main/java/opennlp/tools/util/model/BaseModel.java  | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/opennlp-tools/src/main/java/opennlp/tools/util/model/BaseModel.java b/opennlp-tools/src/main/java/opennlp/tools/util/model/BaseModel.java
index fdae63a89..5fcd1dae6 100644
--- a/opennlp-tools/src/main/java/opennlp/tools/util/model/BaseModel.java
+++ b/opennlp-tools/src/main/java/opennlp/tools/util/model/BaseModel.java
@@ -419,14 +419,11 @@ protected void validateArtifactMap() throws InvalidFormatException {
       // Version check is only performed if current version is not the dev/debug version
       if (!Version.currentVersion().equals(Version.DEV_VERSION)) {
         // Major and minor version must match, revision might be
+        // this check allows for the use of models of n minor release behind current minor release
         if (Version.currentVersion().getMajor() != version.getMajor() ||
-            Version.currentVersion().getMinor() != version.getMinor()) {
-          //this check allows for the use of models one minor release behind current minor release
-          if (Version.currentVersion().getMajor() == version.getMajor() && (Version.currentVersion().getMinor() - 2)
-                  != version.getMinor()) {
+            Version.currentVersion().getMinor() - 2 > version.getMinor()) {
           throw new InvalidFormatException("Model version " + version + " is not supported by this ("
               + Version.currentVersion() +") version of OpenNLP!");
-          }
         }
 
         // Reject loading a snapshot model with a non-snapshot version
